microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,514 @@
+    ).stdout.split()
+    support_kernel_array = re.split('\.|\-', min_support_version)
+    for indx in range(4):
+        if kernel_array[indx] > support_kernel_array[indx]:
+            return True
+        elif kernel_array[indx] < support_kernel_array[indx]:
+            return False
+    return True
+
+
+@TestSuiteMetadata(
+    area="power",
+    category="functional",
+    description="""
+    """
+)
+class power(TestSuite):
+    HB_CUSTOM_KERNEL_URL = "git://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git"
+    HB_CUSTOM_KERNEL_BRANCH = "stable"
+    HB_URL = ""
+
+    @TestCaseMetadata(
+        description="""
+        This test can be performed in Azure and Hyper-V both. But this script only covers Azure.
+        1. Prepare swap space for hibernation
+        2. Compile a new kernel (optional)
+        3. Update the grub.cfg with resume=UUID=xxxx where is from blkid swap disk
+        4. Set RTC to future time or past time
+        5. Hibernate the VM
+        5. Resume the VM
+        6. Read RTC timestamp and compare to the original value
+        """,
+        priority=3
+    )
+    def power_clock_sync_hibernate(self, environment: Environment, node: Node, log: Logger) -> None:
+        max_kernel_compile_min = 90
+        
+        # Prepare the swap space in the target VM
+        resource_group = 
+        vm_name = 
+        location = 
+        storage_type = "StandardSSD_LRS"
+        data_disk_name = vm_name + "_datadisk1"
+
+        is_disk_added = False
+        lun = 0
+        # TODO: CHECK IS_DISK_ADDED
+
+
+        test_command = "echo disk > /sys/power/state"
+        env_setup_result = node.execute(
+            "dmesg | grep -i root= | grep -i resume | wc -l", shell = True, sudo = True
+        )
+        log.debug("Env Setup Result: " + env_setup_result.stdout)
+        if env_setup_result.stdout > "0":
+            log.info("Evironment has been set up...")
+        else:
+            # Configuration for hibernation
+            node.execute(
+                "umask 022"
+            )
+
+            # GetDistro
+            distro = ""
+            # TODO: ADD THIS SECTION
+
+            # GetGuestGeneration
+            exit_code = node.execute(
+                "ls /sys/firmware/efi/"
+            ).exit_code
+            os_generation = 2 if exit_code == 0 else 1
+            
+            if "redhat" in distro and self.HB_URL == "":
+                min_support_kernel = "4.18.0-202"
+                check_vm_support_res = check_vm_feature_support_status(min_support_kernel)
+                current_kernel_version = node.execute(
+                    "uname -r", shell = True, sudo = True
+                )
+                assert_that(
+                    check_vm_support_res,
+                    f"Hibernation is supported since kernel-4.18.0-202. Current version: {current_kernel_version}. Skip the test."
+                ).is_true()
+
+            if isinstance(node.os, Redhat):
```

Comment:
It looks there is no different to Redhat and other distros.

Reply:
linux_path is used for git and compile kernel, removed this